### DISCUSSION TITLE
Custom Icon Sets

### DISCUSSION THREAD

**Discussion by ghost** [kimlimjustin/xplorer#239] [Ideas]
Hey, 
Would it be possible to provide custom icons for folders, filetypes, the Sidebar, etc., for Xplorer?
Could this be an extension like Themes are atm? 

Secondly, could the main Window display the corresponding Icon for folders to improve this? 
Like Downloads shows the "Download" Icon for the Download folder in the Sidebar, but the Main view shows the folder Icon while Documents shows a different Icon. 

I would love to hear your thoughts on this.

**Comment by kimlimjustin** [comment]
> Would it be possible to provide custom icons for folders, filetypes, the Sidebar, etc., for Xplorer?

It will be in the future, in fact, I'm working on it rn.

>Secondly, could the main Window display the corresponding Icon for folders to improve this?
> Like Downloads shows the "Download" Icon for the Download folder in the Sidebar, but the Main view shows the folder Icon while Documents shows a different Icon.

Do you mean on the Download folder, the directories inside it show the icon of the Downloads folder?

**Reply by ghost** [reply]
The Favourites have nice icons while the same folders in the Main View have not.

Additionally, I can't understand why Documents has a different Icon from Downloads in the Main View.

<img width="1002" alt="Bildschirmfoto 2022-02-27 um 12 43 44" src="https://user-images.githubusercontent.com/57394581/155881063-7691c1c5-18f5-4e1a-934c-9fa70bbb12c2.png">

**Reply by kimlimjustin** [reply]
> The Favourites have nice icons while the same folders in the Main View have not.

Just realized it, something went wrong with the library function (fixed this)

> Additionally, I can't understand why Documents has a different Icon from Downloads in the Main View.

This is an intentional design tho, because the favorites show only the icon but the main view indicates a folder (act idk how to explain it), but I think I can add settings choice for this, please open an issue regarding this if you want to

**Reply by ghost** [reply]
> This is an intentional design tho, because the favorites show only the icon but the main view indicates a folder (act idk how to explain it), but I think I can add settings choice for this, please open an issue regarding this if you want to

Okay, it is understandable that the Main View has different Icons than the Sidebar. And instead uses modified Folder icons as some editors use for their file explorers. 
So no Problem here would just be nice to have that common folder like they are in favorites all have a different icon from the standard folder icon.
